### PR TITLE
Run3-alca227 Try to avoid LTO warnings from Calibration/HcalAlCaRecoProducers as reported in the issue #38656

### PR DESCRIPTION
**Pull request by bsunanda**
#### PR description:

Try to avoid LTO warnings from Calibration/HcalAlCaRecoProducers as reported in the issue #38656

#### PR validation:

Use the runTheMatrix test workflows

#### If this PR is a backport please specify the original PR and why you need to backport that PR. If this PR will be backported please specify to which release cycle the backport is meant for:

Nothing special

### CODE REVIEW

**Comment by ChrisMisan** [comment]
@bsunanda please squash code-check commit

**Comment by ChrisMisan** [comment]
type hcal

**Comment by bsunanda** [comment]
@cmsbuild Please test

File: Calibration/HcalAlCaRecoProducers/plugins/AlCaIsolatedBunchSelector.cc [hunk 1]
```diff
@@ -27,25 +27,25 @@
 
 //#define EDM_ML_DEBUG
 
-namespace alCaIsolatedBunchSelector {
+namespace AlCaIsolatedBunch {
```
Comment (tvami) [inline Calibration/HcalAlCaRecoProducers/plugins/AlCaIsolatedBunchSelector.cc:30]: why are you doing this?
https://cms-sw.github.io/cms_coding_rules.html
2.7 says the original was good

Comment (tvami) [reply Calibration/HcalAlCaRecoProducers/plugins/AlCaIsolatedBunchSelector.cc:30]: Or at least the first character of the original was good :)

**Comment by tvami** [comment]
@bsunanda please squash before testing. And also we'll need to do the testing for the LTO release, so the simple `please test` is not enough

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by ChrisMisan** [comment]
please abort

**Comment by ChrisMisan** [comment]
please test for CMSSW_12_5_LTO_X

**Comment by ChrisMisan** [comment]
please test

**Comment by tvami** [comment]
@cmsbuild , please test for CMSSW_12_5_LTO_X

**Comment by tvami** [comment]
Apparently this https://github.com/cms-sw/cmssw/pull/39112#issuecomment-1220705414 didnt trigger it.. weird

**Comment by tvami** [comment]
type bugfix

**Comment by smuzaffar** [comment]
> Apparently this [#39112 (comment)](https://github.com/cms-sw/cmssw/pull/39112#issuecomment-1220705414) didnt trigger it.. weird

if you have more than one test requests then bot only process the last comment. In order to properly request multiple tests one needs to wait for bot to first acknowledge the first request then add the next comment :-)

**Comment by tvami** [comment]
Thanks @smuzaffar that makes sense indeed

**Comment by ChrisMisan** [comment]
@bsunanda I am seeing a lot of differences in https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_12_5_X_2022-08-18-1100+116df4/52228/validateJR.html

**Comment by smuzaffar** [comment]
please test

@ChrisMisan , dqm differences in https://github.com/cms-sw/cmssw/pull/39112#issuecomment-1221009480 are due LTO vs non-LTO comparison. I have requested to re-run the test for default IBs

**Comment by smuzaffar** [comment]
@ChrisMisan , the dqm differences in https://github.com/cms-sw/cmssw/pull/39112#issuecomment-1221009480 are for LTO vs non-LTO relval comparisons. 

Looks like the `addon` job for non-LTO tests was killed that is why the results for non-LTO (default IBs) PR tests are not available yet. I have started the addon job, so you should get the default PR tests results soon

**Comment by smuzaffar** [comment]
by the way, LTO tests shows that warningshas been fixed https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-116df4/26933/build-logs/

**Comment by ChrisMisan** [comment]
> @ChrisMisan , the dqm differences in [#39112 (comment)](https://github.com/cms-sw/cmssw/pull/39112#issuecomment-1221009480) are for LTO vs non-LTO relval comparisons.
> 
> Looks like the `addon` job for non-LTO tests was killed that is why the results for non-LTO (default IBs) PR tests are not available yet. I have started the addon job, so you should get the default PR tests results soon

Thanks @smuzaffar, that makes sense

**Comment by ChrisMisan** [comment]
+alca

- no compilation warnings in LTO/non-LTO
- no differences aside from MTD and msgLogger

**Comment by perrotta** [comment]
+1

**Pull request merged**